File: ulohy/15-desitkovy-rozklad/reseni/main.py
import math


def format_float(number) -> str:
    return ("%f" % number).rstrip("0").rstrip(".")


def print_decomposition(number: int) -> None:
    size = 0
    temp = number
    while temp >= 10:
        temp //= 10
        size += 1

    while number > 0:
        power = math.pow(10, size)
        multiple = number // power

        if multiple != 0:
            print(f"{format_float(multiple)} x {format_float(power)}")

        # Neboli `number = number % power`.
        number %= power
        size -= 1


if __name__ == "__main__":
    number = int(input())

    print_decomposition(number)

File: ulohy/15-desitkovy-rozklad/reseni/main.js
const readline = require("readline");
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
});

rl.question("", input => {
    let number = parseInt(input);
    printDecomposition(number);
    rl.close();
});

function printDecomposition(number) {
    let size = 0;
    let temp = number;
    while (temp >= 10) {
        temp = Math.floor(temp / 10);
        size++;
    }

    while (number > 0) {
        let power = Math.pow(10, size);
        let multiple = Math.floor(number / power);

        if (multiple !== 0) {
            console.log(`${multiple} x ${power}`);
        }

        // Neboli `number = number % power`.
        number %= power;
        size--;
    }
}
